lib/irb/completion.rb

```diff
@@ -33,6 +33,8 @@ def defined? do
       yield
     ]
 
+    HELP_COMMAND_PREPOSING = /\Ahelp\s+/
```

Comment:
I think each command can have a completion and colorization for command arguments (as an internal api).
Do you think we should do it now or not or maybe later?
I'm thinking of something like this
```ruby
if command? && argument_part?
  # pass completor so that command can fallback to completor if the arg part is a ruby code
  return command_instance._internal_complete(command_arg, completor)
end
```

Reply:
IMO, we should first finish completion support for signature arguments too, like `show_source` and `edit`'s, before designing any public API. This is to make sure we know what we want and don't want to expose. I also don't see too strong of a demand for such feature yet, so I think we can wait a bit.